extensions/amp-bind/0.1/bind-impl.js

```diff
@@ -1189,11 +1201,13 @@ export class Bind {
    * @param {Object<string, BindExpressionResultDef>} results
    * @param {Object} opts options bag
    * @param {boolean=} opts.skipAmpState
+   * @param {Array<!Element>=} opts.constrainedEvaluation
    * @return {!Promise}
    * @private
    */
   apply_(results, opts) {
-    const promises = this.boundElements_.map(boundElement => {
+    const elements = opts.constrainedEvaluation || this.boundElements_;
```

Comment:
`this.boundElements_` is an array of `BoundElementsDef` -- check out `applyElements_()`. 

What might be cleanest is merging `applyElements_()` into `apply_()`.

Reply:
Thats a great idea 👍